kroxylicious-systemtests/src/main/java/io/kroxylicious/systemtests/utils/DeploymentUtils.java

```diff
@@ -133,4 +134,22 @@ public static boolean checkLoadBalancerIsWorking(String namespace) {
         kubeClient().getClient().apps().deployments().inNamespace(namespace).withName(TEST_LOAD_BALANCER_NAME).delete();
         return isWorking;
     }
+
+    /**
+     * Wait for run succeeded boolean.
+     *
+     * @param namespaceName the namespace name
+     * @param deploymentName the deployment name
+     * @param timeout the timeout
+     */
+    public static void waitForRunSucceeded(String namespaceName, String deploymentName, Duration timeout) {
+        LOGGER.info("Waiting for Job: {}/{} to be succeeded", namespaceName, deploymentName);
+        with().conditionEvaluationListener(condition -> {
+            if (!condition.isSatisfied() && condition.getRemainingTimeInMS() <= condition.getPollInterval().toMillis()) {
```

Comment:
I think if you implement org.awaitility.core.ConditionEvaluationListener#onTimeout you can do away with the need for the complicated expression.

```
        with().conditionEvaluationListener(new ConditionEvaluationListener<>() {
                    @Override
                    public void onTimeout(TimeoutEvent timeoutEvent) {
                        LOGGER.error("Run failed! Error: {}", kubeClient().logsInSpecificNamespace(namespaceName, deploymentName));
                    }

                    @Override
                    public void conditionEvaluated(EvaluatedCondition condition) {
                        // unused
                    }
                }
```